rest-service/manager_rest/resource_manager.py

```diff
@@ -1499,6 +1505,23 @@ def assert_no_snapshot_creation_running_or_queued(self):
                     ' while a `create_snapshot` workflow is running or queued'
                     ' (snapshot id: {0})'.format(e.id))
 
+    def _cleanup_failed_deployment(self, deployment_id):
```

Comment:
Doesn't this mean that we can't let the user-facing side of this be fully async because the user won't be able to check the failed create_dep_env asynchronously? (obviously it could be fixed later if so)

Reply:
note that this gets called, and so the deployment is removed, on the next attempt to create

that is to say, if you have a failed deployment, you can `cfy dep create xxx` without getting the "deployment by that name already exists"

Reply:
Ah, I misread- sounds good then!